File: lib/features/contents/data/models/category_model.dart
class CategoryModel {
  String? id;
  String? trainerId;
  String? category;
  String? slug;
  String? description;
  bool? isActive;
  String? createdAt;
  String? updatedAt;

  CategoryModel({
    this.id,
    this.trainerId,
    this.category,
    this.slug,
    this.description,
    this.isActive,
    this.createdAt,
    this.updatedAt,
  });

  CategoryModel.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    trainerId = json['trainerId'];
    category = json['category'];
    slug = json['slug'];
    description = json['description'];
    isActive = json['isActive'];
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'trainerId': trainerId,
      'category': category,
      'slug': slug,
      'description': description,
      'isActive': isActive,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
    };
  }
}

File: src/modules/category/category.model.ts
import mongoose, { Schema } from "mongoose";
import { ICategory } from "./category.interface";

const categorySchema = new Schema<ICategory>(
  {
    trainerId: {
      type: Schema.Types.ObjectId,
      ref: "Trainer",
      required: true,
      index: true,
    },
    category: {
      type: String,
      required: true,
    },
    slug: {
      type: String,
      required: true,
    },
    description: {
      type: String,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true },
);

categorySchema.index({ trainerId: 1, slug: 1 }, { unique: true });

export const CategoryModel =
  (mongoose.models.Category as mongoose.Model<ICategory>) ||
  mongoose.model<ICategory>("Category", categorySchema);
